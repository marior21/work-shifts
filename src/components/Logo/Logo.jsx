import { Text, View, styles } from '@mirai/ui';
import PropTypes from 'prop-types';
import React from 'react';

import * as style from './Logo.module.css';

export const Logo = ({ highlight, ...others }) => (
  <View row {...others}>
    <Text bold headline level={highlight ? 3 : 2} className={style.mirai}>
      mro
    </Text>
    <Text headline level={highlight ? 3 : 2} className={styles(style.metasearch, highlight && style.highlight)}>
      workshifts
    </Text>
  </View>
);

Logo.propTypes = {
  highlight: PropTypes.bool,
};
