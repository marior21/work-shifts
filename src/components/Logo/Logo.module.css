.mirai {
  color: var(--mirai-ui-accent);
}

.metasearch {
  color: var(--mirai-ui-content-light);
}

.highlight {
  color: var(--mirai-ui-base);
}

.icon {
  background-color: var(--mirai-ui-accent);
  border-radius: 50%;
  height: var(--mirai-ui-space-L);
  margin-right: var(--mirai-ui-space-XS);
  width: var(--mirai-ui-space-L);
}
