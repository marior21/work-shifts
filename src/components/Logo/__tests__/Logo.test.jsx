import { render } from '@testing-library/react';
import React from 'react';

import { Logo } from '../Logo';

const DEFAULTS = {};

describe('components/ <Logo>', () => {
  test('renders', () => {
    const { asFragment } = render(<Logo {...DEFAULTS} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('prop:highlight', () => {
    const { asFragment } = render(<Logo {...DEFAULTS} highlight />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Testing
  test('testID', () => {
    const { asFragment, getByTestId } = render(<Logo {...DEFAULTS} data-testid="mirai" />);
    expect(getByTestId('mirai')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  // Styling
  test('inherit:className', () => {
    const { asFragment } = render(<Logo {...DEFAULTS} className="mirai" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
