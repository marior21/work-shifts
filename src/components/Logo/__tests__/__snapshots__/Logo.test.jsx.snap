// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ <Logo> inherit:className 1`] = `
<DocumentFragment>
  <div
    class="view row mirai"
  >
    <h2
      class="text bold headline-2"
    >
      mirai
    </h2>
    <h2
      class="text headline-2"
    >
      metasearch
    </h2>
  </div>
</DocumentFragment>
`;

exports[`components/ <Logo> prop:highlight 1`] = `
<DocumentFragment>
  <div
    class="view row"
  >
    <h3
      class="text bold headline-3"
    >
      mirai
    </h3>
    <h3
      class="text headline-3"
    >
      metasearch
    </h3>
  </div>
</DocumentFragment>
`;

exports[`components/ <Logo> renders 1`] = `
<DocumentFragment>
  <div
    class="view row"
  >
    <h2
      class="text bold headline-2"
    >
      mirai
    </h2>
    <h2
      class="text headline-2"
    >
      metasearch
    </h2>
  </div>
</DocumentFragment>
`;

exports[`components/ <Logo> testID 1`] = `
<DocumentFragment>
  <div
    class="view row"
    data-testid="mirai"
  >
    <h2
      class="text bold headline-2"
    >
      mirai
    </h2>
    <h2
      class="text headline-2"
    >
      metasearch
    </h2>
  </div>
</DocumentFragment>
`;
